src/components/Select/Select.tsx

```diff
@@ -151,7 +151,12 @@ export const Select: React.FC<SelectProps> = ({
         aria-labelledby={`Select__label--${idRef} Select__value--${idRef}`}
         type='button'
       >
-        <label id={`Select__label--${idRef}`}>{label}</label> {(focused ?? selected)?.label}
+        {label && (
+          <label id={`Select__label--${idRef}`}>{`${label}${
+            state.value !== undefined ? ':' : ''
+          }`}</label>
+        )}{' '}
```

Comment:
I would create a component for this, within `<Select />`s folder, something like `<SelectLabel />` to increase the readability of this.